### PR TITLE
Feature/implement form builder

### PR DESCRIPTION
**Pull request by hesamhabibih**

### CODE REVIEW

File: components/CoreComponents/CheckBox/CheckBox.tsx [hunk 1]
```diff
@@ -7,30 +7,31 @@ const Checkbox = (CheckboxProps: ICheckboxProps) => {
...
           checked={isChecked}
           disabled={isDisabled}
           readOnly={isReadonly}
+          required={isRequired}
+          aria-invalid={ariaInvalid}
           id={label}
           name={label}
-          onChange={(e) => e.target.value}
+          // onChange={(e) => e.target.value}
+          onChange={(e) => CheckboxProps.onChange?.(e)}
```
Comment (saralabafi) [inline components/CoreComponents/CheckBox/CheckBox.tsx]: add onChange  to CheckboxProps

File: components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx [hunk 2]
```diff
@@ -61,6 +60,7 @@ const SwitchControl: React.FC<SwitchControlProps> = ({
                 disabled={disabled}
                 onChange={(checked: boolean) => {
                   // Handle the toggle change logic
+                  console.log('Toggle changed:', checked)
```
Comment (saralabafi) [inline components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx]: remove console.log

File: components/CoreComponents/CheckBox/CheckBox.tsx (hunk 1 above)
Comment (hesamhabibih) [reply components/CoreComponents/CheckBox/CheckBox.tsx]: ok edited it

File: components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx (hunk 2 above)
Comment (hesamhabibih) [reply components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx]: ok

File: components/CoreComponents/CheckBox/CheckBox.tsx [hunk 3]
```diff
@@ -7,30 +7,31 @@ const Checkbox = (CheckboxProps: ICheckboxProps) => {
...
+        className={`${backgroundColor}  ${borderColor} ${textColor} ${direction}`}>
         <label htmlFor={label}>{label}</label>
         <input
           className={`${direction}`}
           type="checkbox"
           checked={isChecked}
           disabled={isDisabled}
           readOnly={isReadonly}
+          required={isRequired}
+          aria-invalid={ariaInvalid}
```
Comment (amirkh330) [inline components/CoreComponents/CheckBox/CheckBox.tsx]: remove it, please

File: components/CoreComponents/CheckBox/CheckBox.type.tsx [hunk 4]
```diff
@@ -4,13 +4,15 @@ export interface ICheckboxProps {
   isChecked: boolean
   isDisabled?: boolean
   isReadonly?: boolean
+  isRequired?: boolean
+  ariaInvalid?:any
```
Comment (amirkh330) [inline components/CoreComponents/CheckBox/CheckBox.type.tsx]: remove it, please

File: components/CoreComponents/Switch/Switch.tsx [hunk 5]
```diff
@@ -1,81 +1,52 @@
...
+        onClick={handleClick}>
+        {/* Switch */}
+        <div
+          className={`bg-blue-500 md:w-4 md:h-4 h-3 w-3 rounded-full shadow-md transform duration-300 ease-in-out${
+            toggle ? '' : toggleClass
+          }`}></div>
+        {/* Tick */}
+        {toggle && (
+          <></>
+          //  <svg></svg>
```
Comment (amirkh330) [inline components/CoreComponents/Switch/Switch.tsx]: use image

File: components/CoreComponents/Switch/Switch.types.ts [hunk 6]
```diff
@@ -1,16 +1,11 @@
...
-  checkedColor?: 'primary' | 'error' | 'success' 
-  ref?: React.Ref<HTMLInputElement> | null
-  onChange?: () => void
-  edge?: 'start' | 'end'
+export interface SwitchProps {
   id?: string
+  type?: string
+  className?: string
+  disabled?: boolean
+  onChange?: any
```
Comment (amirkh330) [inline components/CoreComponents/Switch/Switch.types.ts]: use correct type

File: components/CoreComponents/Switch/Switch.types.ts [hunk 7]
```diff
@@ -1,16 +1,11 @@
...
+  checked?: boolean
   required?: boolean
-  size?: 'small' | 'medium'
-  value?: any
+  ariaInvalid?: any
```
Comment (amirkh330) [inline components/CoreComponents/Switch/Switch.types.ts]: remove it

File: components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx [hunk 8]
```diff
@@ -1,22 +1,25 @@
-import Switch from './Switch'
+// import Switch from './Switch'
```
Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx]: remove useless Line

File: components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx [hunk 9]
```diff
@@ -1,22 +1,25 @@
...
-  onChange,
+  onChangeSwitch,
   disabled,
   isRequired,
   errorMessage,
   helpText,
+  type,
 }) => {
   const [isChecked, setIsChecked] = useState(isCheckedControl)
   const [isCheckbox, setIsCheckbox] = useState(false)
```
Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx]: move these lines in biz file
SwitchControl.biz.ts

**Pull request merged**